packages/wallets/src/api/client.ts

```diff
@@ -1,4 +1,4 @@
-import { type Crossmint, CrossmintApiClient } from "@crossmint/common-sdk-base";
+import { type Crossmint, APIKeyUsageOrigin, CrossmintApiClient, validateAPIKey } from "@crossmint/common-sdk-base";
```

Comment:
Here, we're making the API client more high-level by:

1. Aligning client- and server-side endpoint methods to have a similar signature
2. Switching between those endpoints based on the API key

That way, the consumer of `ApiClient` don't need to worry about that at all!